dask/array/linalg.py

```diff
@@ -183,10 +185,12 @@ def tsqr(data, compute_svd=False, _max_vchunk_size=None):
                                            [(name_r_st1, idx, 0)
                                             for idx, _ in sub_block_info])))
                              for i, sub_block_info in enumerate(all_blocks))
-        dsk.update_with_key(dsk_r_stacked, key=name_r_stacked)
+        dsk.update_with_key(dsk_r_stacked, key=name_r_stacked,
+                            dependencies={name_r_st1})
 
         # retrieve R_stacked for recursion with tsqr
         vchunks_rstacked = tuple([sum(map(lambda x: x[1], sub_block_info)) for sub_block_info in all_blocks])
+        dsk.dependencies[name_r_stacked] = set(data.name)
```

Comment:
Is `data.name` a string? Should this instead be `set([data.name])`?

Reply:
Or rather `{data.name}`

Reply:
Thank you for the catch.  This was, indeed, a bug.  

I've added an assert statement for now to catch these in the future.  I suspect that there will be some churn in this area soonish.